### PR TITLE
fix: assign default validation method if blank before save

### PR DESCRIPTION
**Pull request by hitenshKharva**
Ref: [Asana Task](https://app.asana.com/0/1199082161983365/1199608531478425)

Default Validation Method Set: Moving Average 
image

https://user-images.githubusercontent.com/56474017/110122292-72af3580-7de5-11eb-94c6-ffeeb5711722.mp4

### CODE REVIEW

File: erpnext/stock/doctype/item/item.py [hunk 1]
```diff
@@ -889,6 +890,11 @@ def validate_auto_reorder_enabled_in_stock_settings(self):
 			if not enabled:
 				frappe.msgprint(msg=_("You have to enable auto re-order in Stock Settings to maintain re-order levels."), title=_("Enable Auto Re-Order"), indicator="orange")
 
+	def validate_valuation_method(self):
+			if not self.valuation_method:
+				method = frappe.db.get_single_value("Stock Settings", "valuation_method") or "FIFO"
+				self.valuation_method = method
```
Comment (hrwX) [inline erpnext/stock/doctype/item/item.py]: - Set the default in Stock Settings
- Before saving stock settings, if empty set FIFO in there

**Review by hrwX — CHANGES_REQUESTED** [review]

**Review by rvpasari — DISMISSED** [review]

**Review by hrwX — APPROVED** [review]

File: erpnext/stock/doctype/stock_settings/stock_settings.js [hunk 2]
```diff
@@ -13,5 +13,11 @@ frappe.ui.form.on('Stock Settings', {
 
 		frm.set_query("default_warehouse", filters);
 		frm.set_query("sample_retention_warehouse", filters);
+	},
+	
+	before_save: function(frm){
+		if (!frm.doc.valuation_method){
+			frm.set_value("valuation_method", "FIFO");
+		}
```
Comment (hrwX) [inline erpnext/stock/doctype/stock_settings/stock_settings.js]:
```suggestion
	before_save: function(frm){
		if (!frm.doc.valuation_method){
			frm.set_value("valuation_method", "FIFO");
		}
```
Set FIFO as default in stock settings json

**Review by hrwX — CHANGES_REQUESTED** [review]

**Comment by sahil28297** [comment]
@hitenshKharva you'll need to add a patch as well to set it in stock_settings

**Comment by hitenshKharva** [comment]
@sahil28297 added patch

**Review by hrwX — APPROVED** [review]

**Comment by sahil28297** [comment]
@hitenshKharva add proper task link

**Comment by hitenshKharva** [comment]
@sahil28297 the link was copied wrong by mistake. Corrected it.

**Pull request merged**